### PR TITLE
Generate rate limit frontend api handler

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Generate rate limit frontend api handler code
Some fix:
- Change rate limit type of the following APIs:
- ScanWorkflowExecution
- RefreshWorkflowTasks
- GetClusterInfo

**Why?**
To reduce manual work

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/frontend/templates/ratelimited.tmpl
Comment (Groxx) [inline service/frontend/templates/ratelimited.tmpl:95]: fwiw: you can indent the code lines on all of these as deeply as you want, the formatting step will fix the indentation.

it won't remove empty lines, so the `{{- ` is still useful/desirable, but indents help a fair bit with where the if/ends are paired.

File: service/frontend/templates/ratelimited.tmpl
Comment (Groxx) [inline service/frontend/templates/ratelimited.tmpl:38]: I'll vote to group by type and alphabetize within that group fwiw.  makes it easier to spot associations (which are pretty relevant).

**Review by Groxx — COMMENTED** [review]
gonna stick myself in here for "I think this needs a careful read before approval, but broadly I like it".

main concern is that this makes a significant change to when the call occurs, and that needs to be verified to be safe.

whether we return a 429 or "bad request" error on floods of bad requests: eh, I think both are defensible, changing it seems fine to me.  it's just worth checking if there are any particularly valuable ones that will be lost (I sincerely doubt it).

File: service/frontend/service.go
Comment (Groxx) [inline service/frontend/service.go:166]: I think I approve of this change, just making it visible / as a reminder for myself for later: this will mean that ratelimited requests no longer contribute to response latency / success rate.

File: service/frontend/templates/ratelimited.tmpl
Comment (Groxx) [inline service/frontend/templates/ratelimited.tmpl:104]:
```
            {{- $rateLimitHelper := "allowDomainID"}}
```
with `:=`, this looks unused in this block.  or does it still modify the lines below that have this?
```
            h.{{$rateLimitHelper}}({{$ratelimitType}}, {{$domain}})
```

(I'm not 100% sure what the scoping behavior is with templates, but I assume it's the same as go...?  hopefully?)

File: service/frontend/wrappers/ratelimited/ratelimit.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/ratelimit.go:49]: this is a fairly minimal change and if you prefer that then 👍 but:
we could take this opportunity to remove this method and directly call the appropriate limiter.

I didn't do that initially when adding these, to keep things minimal, but it's definitely due for a cleanup at some point.  Does not have to be now though.

File: service/frontend/templates/ratelimited.tmpl
Comment (Shaddoll) [reply service/frontend/templates/ratelimited.tmpl:104]: it's unused

File: service/frontend/api/handler.go
Comment (Groxx) [inline service/frontend/api/handler.go:501]: so far I think the only concerning point is that this check is done earlier:
```go
	domainName := pollRequest.GetDomain()
	if domainName == "" {
		return nil, validate.ErrDomainNotSet
	}
```
which means that after this change, `wh.allow` may be called with an empty domain string.

I'm not 100% sure what happens in that case tbh.  I doubt it's _damaging_, so it might be fine to do that, but it's worth figuring out.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:200]: probably missing an empty-check / the `validate.ErrDomainNotSet` branch.  the domain cache doesn't seem to check for empty values early on at least, kinda seems like it might not expect emptys.

we might also want to change the deserialize-err to a bad-request too, but technically that'd be a semantic change / only if you feel like it IMO.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:372]: surprisingly, ScanWorkflowExecutions is currently classified as a "user" request.

that does seem wrong, so 👍 to this change, just calling it out for visibility.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:68]: same as scan, surprisingly this was user too, also 👍

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:100]: previously user-limited, but this seems fine.  it's a trivial endpoint / other cluster-wide things don't have limits and nobody really calls them anyway.

**Review by Groxx — COMMENTED** [review]
All done with a round-1 careful read!

A couple smallish changes worth verifying, but other than those / check comments, this all looks good.  should be pretty easy to re-review after those things are checked/changed.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:223]: ah, great, good catch.  yes, this should be user-limited.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [reply service/frontend/wrappers/ratelimited/api_generated.go:200]: yea, domain cache will go this all the way down to the database without checking.  it's worth checking here.

File: service/frontend/wrappers/ratelimited/api_generated.go
Comment (Groxx) [inline service/frontend/wrappers/ratelimited/api_generated.go:417]: at some point we should probably make some validation-wrappers that occur early and get rid of this need at random levels...
but 👍 yep, same as existing behavior, LGTM

**Review by Groxx — APPROVED** [review]
Yep, looks good!  Thanks for the template-ization, the more we can standardize and pull out of the handlers the better 👍

**Pull request merged**
